### PR TITLE
Add CI/CD

### LINKED ISSUE #4: Setup CI/CD with github actions to lint and test branches
Good starting point for GH Actions for PRs is in https://github.com/jonhoo/rust-ci-conf

Probably should also take a look at axodotdev stuff, mainly [cargo-dist](https://github.com/axodotdev/cargo-dist)

### PR DESCRIPTION
**Pull request by mikhailantoshkin**
Closes #4 
This PR adds CI/CD pipelines from [cargo-dist](https://github.com/axodotdev/cargo-dist) for releases and [jonhoo](https://github.com/jonhoo/rust-ci-conf) for checking PRs.

For release, it's quite straightforward. Just tag the `main` branch, push it to remote and voilà - you have a release. For specifics, refer to the [cargo-dist book](https://opensource.axo.dev/cargo-dist/book/way-too-quickstart.html).

PRs checking is a copy of jonhoo's CI, with some minor tweaks. Notably: ignoring nightly rust and msrv. CI for tests will be added when there will be tests to run)

### CODE REVIEW

**Comment by mikhailantoshkin** [comment]
Oh no, it was not `cargo fmt`'ed properly!

**Comment by mikhailantoshkin** [comment]
Also, there are some clippy warnings, I suggest we address them at some point

File: Cargo.toml [hunk 1]
```diff
@@ -17,3 +17,19 @@ name = "bfint-cli"
...
+# Config for 'cargo dist'
+[workspace.metadata.dist]
+# The preferred cargo-dist version to use in CI (Cargo.toml SemVer syntax)
+cargo-dist-version = "0.0.5"
+# The preferred Rust toolchain to use in CI (rustup toolchain syntax)
+rust-toolchain-version = "1.68.0"
+# CI backends to support (see 'cargo dist generate-ci')
+ci = ["github"]
+# Target platforms to build apps for (Rust target-triple syntax)
+targets = ["x86_64-unknown-linux-gnu", "x86_64-apple-darwin", "x86_64-pc-windows-msvc", "aarch64-apple-darwin"]
```
Comment (WindowGenerator) [inline Cargo.toml:35]: I think we can add `aarch64-unknown-linux-gnu` in the future

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  create-release:
+    runs-on: ubuntu-latest
+    outputs:
+      has-releases: ${{ steps.create-release.outputs.has-releases }}
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Rust
+        run: rustup update 1.68.0 --no-self-update && rustup default 1.68.0
```
Comment (WindowGenerator) [inline .github/workflows/release.yml:55]: Wow, rustup appeared in the latest ubuntu 
image

**Comment by WindowGenerator** [comment]
Can u add yourself in authors btw
[here](https://github.com/WindowGenerator/bfint/pull/10/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R3)

**Review by WindowGenerator — APPROVED** [review]
Great job!
Thx for contributing!

**Pull request merged**